### ISSUE TITLE
uboot fails to load kernel on raspi 4B

### ISSUE THREAD

**Issue by EasyIP2023** [rauc/meta-rauc-community#21]
I cant seem to get the PI to boot with A/B dual partition + rauc.  I've followed this tutorial here https://www.konsulko.com/getting-started-with-rauc-on-raspberry-pi-2/, but no matter what I do I always see the bellow error. Even setting ```kernel=uImage``` in ```config.txt``` and setting ```KERNEL_IMAGETYPE = "uImage"``` yields the same results.

![rauc-error](https://user-images.githubusercontent.com/8561172/131790238-b49b5540-a46d-41e6-a748-64bf34a089e4.png)

**Comment by leon-anavi** [comment]
Hi @EasyIP2023,

Which Yocto/OE release and respectively branch of `meta-rauc-community` are you using?

Keep in mind that the project has evolved since I wrote the article and numerous community contributions have been merged in `master` and honestly due to lack of enough time for testing they may resulted in certain issues.

Thanks,
Leon

**Comment by EasyIP2023** [comment]
@leon-anavi no worries. Thanks for writing that! I was on Yocto/OE (Dunfell 3.1.8) when I was first testing later switched to (Dunfell 3.1.10). Also on the latest commit of meta-rauc-community branch dunfell. Will first try to switch to branch master. If that doesn't work revert to a later meta-rauc-community commit maybe one from before Sept 2020.

**Comment by leon-anavi** [comment]
Hi @EasyIP2023,

Sounds as a good plan. If you are using Yocto/OE release Dunfell 3.1.8 I would recommend using the `dunfell` branch. There were some recent syntax changes in `master` which are not compatible with `dunfell`, for example GitHub pull requests https://github.com/rauc/meta-rauc-community/pull/20 and https://github.com/rauc/meta-rauc-community/pull/19.

Please provide feedback of your finding as it will be useful for the community and fixing these issues.

Thanks,
Leon

**Comment by EasyIP2023** [comment]
@leon-anavi Closing issue as after some investigation the problem isn't with the meta-rauc-community/meta-rauc-raspberrypi layer, but with u-boot support for the raspberrypi 4B. Seems like it doesn't want to work with the latest raspberrypi bootloader/firmwar.

```
U-Boot 2021.10-rc3-dirty (Sep 05 2021 - 15:23:39 -0500)

DRAM:  3.9 GiB
RPI 4 Model B (0xc03112)
mbox: Timeout waiting for response
bcm2835: Could not set module 3 power state
initcall sequence 000000003b3d28f8 failed at call 0000000000084344 (err=-5)
```

Here's a link to something similar years ago. Implemented by agherzan.

* https://www.raspberrypi.org/forums/viewtopic.php?t=244785

I decided to try the compile u-boot from source and try it on the latest raspberry pi stock image. Ran into similar results. Which leads me to believe. u-boot isn't configured to work with the latest raspberrypi bootloader/firmware. At least from what I am seeing.

* https://andrei.gherzan.ro/linux/uboot-on-rpi/
